docs/types/reference-types.rst

```diff
@@ -336,6 +336,53 @@ Array Members
         }
     }
 
+.. index:: ! array;slice
+
+.. _array-slices:
+
+Array Slices
+------------
+
+
+Array slices are a view on a contigous portion of an array.
+They are written as ``x[start:end]``, where ``start`` and
+``end`` are expressions resulting in a uint256 type (or
+implicitly convertible to it). The first element of the
+slice is ``x[start]`` and the last element is ``x[end - 1]``.
+
+If ``start`` is greater than ``end`` or if ``end`` is greater
+than the length of the array, an exception is thrown.
+
+Both ``start`` and ``end`` are optional, and they default to
+``0`` and the length of the array, respectively.
+
+Array slices do not have any members. They are implicitly
```

Comment:
Well the "implicitly convertible to arrays of their underlying type" is true for calldata arrays - but I'm not yet sure what will happen for memory or storage... but we can adjust this once that's being worked on... maybe we'll have ``copyof`` by then, then that will be needed for the conversion...

Reply:
Okay, if I understood correctly, we leave this as it is for now, changing it as required when an implementation for memory/storage array slices is available.